### PR TITLE
provider/openstack: lbaasv2 caching and provider complete

### PR DESCRIPTION
**Pull request by drmaas**

### CODE REVIEW

**Comment by drmaas** [comment]
@spinnaker/reviewers @derekolk @varikin @emjburns @ashleykasim

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/controllers/OpenstackLoadBalancerController.groovy [hunk 1]
```diff
@@ -42,12 +42,13 @@ class OpenstackLoadBalancerController {
   @RequestMapping(method = RequestMethod.GET)
   Set<OpenstackLoadBalancerSummary> list() {
     provider.getLoadBalancers('*', '*', '*').collect { lb ->
+      println lb.toString()
```
Comment (derekolk) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/controllers/OpenstackLoadBalancerController.groovy]: Remove println

File: clouddriver-openstack/clouddriver-openstack.gradle [hunk 2]
```diff
@@ -19,8 +19,8 @@ repositories {
 }
 
 ext {
-  openstack4jCoreVersion = '3.0.2-20160822.115544-6'
-  openstack4jJerseyVersion = '3.0.2-20160822.115557-6'
+  openstack4jCoreVersion = '3.0.2'
```
Comment (varikin) [inline clouddriver-openstack/clouddriver-openstack.gradle]: Didn't @derekolk's PR yesterday handle this?

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/model/OpenstackLoadBalancer.groovy [hunk 3]
```diff
@@ -39,45 +41,80 @@ class OpenstackLoadBalancer implements LoadBalancer, Serializable, LoadBalancerR
...
-
-  Integer getInternalPort() {
-    parseInternalPort(description)
+    Set<OpenstackLoadBalancerListener> openstackListeners = listeners?.collect { listener ->
+      new OpenstackLoadBalancerListener(externalProtocol: listener.protocol.toString(),
+        externalPort: listener.protocolPort.toString(),
+        description: listener.description)
+    }?.toSet() ?: [].toSet()
+    OpenstackHealthMonitor openstackHealthMonitor = healthMonitor ? new OpenstackHealthMonitor(id: healthMonitor.id,
+      adminStateUp: healthMonitor.adminStateUp, delay: healthMonitor.delay, maxRetries: healthMonitor.maxRetries,
```
Comment (lwander) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/model/OpenstackLoadBalancer.groovy]: Are you positive `healthMonitor` will be set?

**Comment by lwander** [comment]
LGTM after comment

File: clouddriver-openstack/clouddriver-openstack.gradle (hunk 2 above)
Comment (drmaas) [inline clouddriver-openstack/clouddriver-openstack.gradle]: I'll rebase and fix

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/model/OpenstackLoadBalancer.groovy (hunk 3 above)
Comment (drmaas) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/model/OpenstackLoadBalancer.groovy]: It can be null.

Comment (lwander) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/model/OpenstackLoadBalancer.groovy]: Sorry - missed that! LGTM

**Pull request merged**